modules/core/02-client/keeper/migrations.go

```diff
@@ -31,3 +32,17 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 func (m Migrator) Migrate3to4(ctx sdk.Context) error {
 	return v7.MigrateLocalhostClient(ctx, m.keeper)
 }
+
+// MigrateParams migrates from consensus version 4 to 5.
+// This migration takes the parameters that are currently stored and managed by x/params
+// and stores them directly in the ibc module's state.
+func (m Migrator) MigrateParams(ctx sdk.Context) error {
```

Comment:
Talked a bit with Carlos and he mentioned structuring this migration as we do for `v7`. Though it is doable, it requires a separate `expected_keepers.go` and some restructuring in order to pass the `legacySubspace` to the migration.

Thoughts?

Reply:
I don't see why you need a separate `expected_keepers.go`